doc/new-guidelines.md

```diff
@@ -80,6 +80,63 @@ This mistake mostly happens with functional components
 
 it's important to name functional components with `f-` prefix
 
+### Component props and api scheme to match Figma as closely as possible
+
+Ideally, the prop names for components (particularly in quo2 Design System) should match the Figma properties as best 
+as possible. This makes it easier for the developer using that component to then configure it correctly for the screen it is
+being used on and avoids unnecessary overwrites and adjustments being made.
+
+Some examples include
+
+#### Avoid unnecessarily grouping categories to reduce the number of props
+For example in Figma if there is component and it has variants the following variants
+
+|theme: "light" blur: "False"|theme: "dark" blur: "False"|theme: "light" blur: "True"|theme: "dark"  blur: "True"|
+|----------------------------|---------------------------|---------------------------|---------------------------|
+| type :neutral name "ABC"   | type :neutral name "ABC"  |                           |                           |
+| type :active name "ABC"    | type :active name "ABC"   |                           |                           |
+| type :danger name "ABC"    | type :danger name "ABC"   | type :danger name "ABC"   | type :danger name "ABC"   |
+
+```clojure
+;; bad
+"theme - :light or :dark
+type - can be :neutral :active :danger :danger-blur"
+(defn my-component [{:keys [theme type]} name])
+
+;; good
+"theme - :light or :dark
+ type - can be :neutral :active :danger
+ blur? - boolean
+"
+(defn my-component [{:keys [theme blur? type]} name])
+```
+
+Please note this is only for the external api of the component and there should be no restriction of how the component manages
```

Comment:
I remember someone recommended using collapsible sections to explain the why in more detail, which I quite like tbh. We can achieve this with something like:

```
<details>
  <summary>Explanation</summary>
</details>
```

It's interesting if we start to write the text with a TL;DR style plus a longer version when necessary (collapsed by default).

Reply:
will try it out 👍

Reply:
@ilmotta I'm looking at this feature and imo it will make the docs much much easier to read, especially since they have grown quite large. However from experimenting with it I feel it is a bit confusing for one example to use this approach. 
I will instead suggest that I can refactor the docs using this approach as a follow up and we can see how it works across all examples in this doc. I think it will make the doc much easier to parse 👍

Reply:
I see, I felt the same when I tried to use this feature. It requires some rewriting in the docs to make everything easy to read. Thanks @J-Son89